src/elements/hx-file-icon/index.spec.js

```diff
@@ -0,0 +1,154 @@
+            });
+        });
+
+        describe('verify Shadow DOM markup', () => {
+            it('markup should contain a #hxFileIcon <div>', async () => {
+                const elSelector = 'div#hxFileIcon';
+                const id = 'hxFileIcon';
+                const component = /** @type { HXFileIconElement } */ await fixture(template);
+                const shadow = component.shadowRoot;
+                const query = shadow.querySelector(elSelector);
+                const queryId = query.id;
+
+                expect(queryId).to.equal(id);
+            });
+
+            it('markup should contain a #hxBase with an file type <hx-icon>', async () => {
+                const elSelector = 'div#hxFileIcon > hx-icon';
+                const Type = 'file';
+                const component = /** @type { HXFileIconElement } */ await fixture(template);
+                const shadow = component.shadowRoot;
+                const query = shadow.querySelector(elSelector);
+                const queryId = query.type;
+
+                expect(queryId).to.equal(Type);
+            });
+
+            it('markup should contain a #hxOverlay <div>', async () => {
+                const elSelector = 'div#hxOverlay';
+                const id = 'hxOverlay';
+                const component = /** @type { HXFileIconElement } */ await fixture(template);
+                const shadow = component.shadowRoot;
+                const query = shadow.querySelector(elSelector);
+                const queryId = query.id;
+
+                expect(queryId).to.equal(id);
+            });
+
+            it('markup should contain a #hxIcon <hx-icon>', async () => {
+                const elSelector = 'div#hxOverlay > #hxIcon';
+                const id = 'hxIcon';
+                const component = /** @type { HXFileIconElement } */ await fixture(template);
+                const shadow = component.shadowRoot;
+                const query = shadow.querySelector(elSelector);
+                const queryId = query.id;
+
+                expect(queryId).to.equal(id);
+            });
+
+            it('markup should contain a #hxExt <div>', async () => {
+                const elSelector = 'div#hxExt';
+                const id = 'hxExt';
+                const component = /** @type { HXFileIconElement } */ await fixture(template);
+                const shadow = component.shadowRoot;
+                const query = shadow.querySelector(elSelector);
+                const queryId = query.id;
+
+                expect(queryId).to.equal(id);
+            });
+        });
+    });
+
+    describe('test $onConnect method', () => {
+        it('should be type on connect', async () => {
+            const component = /** @type {HXFileIconElement} */ await fixture(template);
+            const attr = component.hasAttribute('type');
+
+            expect(attr).to.be.false;
+        });
+    });
+
+    describe('test <hx-file-icon> getter and setter methods', () => {
```

Comment:
Please add a test for setting the `extension` attribute.

Reply:
Added the extension attribute test case , Please review